docs/eeps/draft-embulk-system-properties-and-embulk-home.md

```diff
@@ -0,0 +1,144 @@
+
+Some performance-related options, especially parameters across plugins, are system-wide. For example, `max_threads`, `min_output_tasks`, and `page_size` were configured in System Config.
+
+Requirements of the New System-wide Configuration
+==================================================
+
+We wanted an alternative system-wide configuration because System Config was not functioning effectively. The requirements would be as below.
+
+* It can be loaded from a file, not only from command-line options.
+* It can be configured from the very beginning of an Embulk process.
+* The configuration file on the file system should be identified by a deterministic and user-friendly rule.
+* The plugin installation directories on the file system should be identified by a deterministic and user-friendly rule.
+* It does not need so powerful representation. A simple string-string key-value mapping would be sufficient.
+
+Embulk System Properties
+=========================
+
+The new system-wide configuration is designed as below, to satisfy the requirements above.
+
+* The new system-wide configuration is based on [`java.util.Properties`](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html), named "Embulk System Properties". `Properties` is not very rich, indeed. But, `Properties` can be loaded from a file only with the Java standard class library in the very beginning of a Java process.
+* A new concept of the "Embulk Home" directory is introduced. The Embulk Home directory takes over the older `~/.embulk/` directory that contained plugin installations. The Embulk Home directory includes the Embulk System Properties file, and continues to have plugin installations by default.
+* The Embulk Home directory and the Embulk System Properties are configured by a deterministic rule at runtime, based on the command-line options, the environment variables, the Embulk System Properties file, and the current working directory.
+
+The details follow.
+
+Setting the Embulk System Properties
+-------------------------------------
+
+The Embulk System Properties are set by the following methods, in priority from top to bottom.
+
+1. The command-line option `-X`, for example, `-Xproperty=value`.
+2. The Embulk System Properties file in the [Java `.properties` format](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html). The location of the Embulk System Properties file is identified based on the Embulk Home directory.
+
+Embulk Home
+------------
+
+The Embulk Home directory is determined at runtime, by the following rule, in priority from top to bottom.
+
+1. If an Embulk System Property `embulk_home` is set by the command-line option `-X`, it is the Embulk Home directory.
+    * It would be an absolute path, or a relative path from the current working directory.
+    * Ex. `java -jar embulk.jar -Xembulk_home=/var/tmp/foo run ...` (absolute path)
+    * Ex. `java -jar embulk.jar -Xembulk_home=.embulk2/bar run ...` (repative path)
```

Comment:
repative  -> relative path

Reply:
Fixed it along with b66db914ce7facb66744d3167d76920db178f31b